### ISSUE TITLE
--worker_extra_flag does not respect worker-key-mnemonic

### ISSUE THREAD

**Issue by AlessandroPatti** [bazelbuild/bazel#24237]
### Description of the bug:

The `worker-key-mnemonic` execution info allows sharing the same worker across multiple actions using different mnemonics by overriding the mnemonic that identifies the worker. This allows avoiding spawning multiple, usually resource intensive, workers for different actions that use the same worker executable. However, this does not work with `--worker_extra_flag=<mnemonic>=<flag>` since the mnemonic in this syntax is not the worker mnemonic, but the action mnemonic. This results in spawning multiple worker, nullifying the effect of the execution info, unless the same flags are specified for all the mnemonic using the worker.

### Which category does this issue belong to?

Rules API

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Create a rule with two actions:
```
ctx.actions.run(
    mnemonic = "Foo",
    executable = ctx.executable._worker,
    execution_info = {"worker-key-mnemonic": "Foo"},
    ...
)
ctx.actions.run(
    mnemonic = "Bar",
    executable = ctx.executable._worker,
    execution_info = {"worker-key-mnemonic": "Foo"},
    ...
)
```
The build a target of that rule passing `--worker_extra_flag=Foo=--bar` and see two workers spawned.

### Which operating system are you running Bazel on?

Ubuntu

### What is the output of `bazel info release`?

release 6.5.0

**Comment by tjgq** [comment]
@bazel-io fork 8.0.0

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 8.0.0 RC3](https://github.com/bazelbuild/bazel/releases/tag/8.0.0rc3). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=8.0.0rc3. Thanks!
